L1Trigger/L1THGCalUtilities/plugins/selectors/HGC3DClusterSimpleSelector.cc

```diff
@@ -0,0 +1,52 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/L1THGCal/interface/HGCalMulticluster.h"
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+
+namespace l1t {
+    class HGC3DClusterSimpleSelector : public edm::stream::EDProducer<> {
```

Comment:
looks like this could be `global`

Reply:
Do you mean the class could be defined outside the `l1t` namespace?

Reply:
No, it could be `edm::global::EDProducer<>` rather than `edm::stream::EDProducer<>`.

Reply:
This module processes each event individually and doesn't need to see all events from the source, so it should be a `stream` module, no?

Reply:
A `stream` module is instantiated once per thread, so "stream scratch" can be used (changing member variable values in the `produce`) function. If "stream scratch" is not needed, it is better to use a `global` module, which is instantiated only once, with all events run through the `const` `produce` function in a functional manner.